experimental/filesystem-backends.mdx

```diff
@@ -0,0 +1,332 @@
+
+```yaml
+version: "1.0"
+include:
+  - "**/features.yaml"
+  - "**/features.yml"
+  - "**/*.features.yaml"
+  - "**/*.features.yml"
+exclude: []
+```
+
+The index file contains two lists `include` and `exclude`. These can contain specific paths or Glob matching patterns.
+The indexing process first matches the `include` section and then filters that set by the `exclude` section.
+
+### Defining Flag State
+
+<Info>
+  Flipt flag state file format has been taken directly from Flipts existing
+  [import and export](/configuration/storage#import-export) flag state format.
+  You can run `flipt export` and add / commit the result to a directory or git
+  repository to get started quickly.
+</Info>
+
+```yaml features.yaml
+namespace: backend
+flags:
+  - key: awesomeNewFeature
+    name: Awesome New Feature
+    enabled: true
+    variants:
+      - key: enabled
+        name: Enabled
+      - key: disabled
+        name: Disabled
+    rules:
+      - segment: internal-users
+        rank: 1
+        distributions:
+          - variant: enabled
+            rollout: 100
+      - segment: all-users
+        rank: 2
+        distributions:
+          - variant: enabled
+            rollout: 20
+          - variant: disabled
+            rollout: 80
+
+segments:
+  - key: internal-users
+    name: Internal Users
+    constraints:
+      - type: STRING_COMPARISON_TYPE
+        property: organization
+        operator: eq
+        value: internal
+    match_type: ALL_MATCH_TYPE
+  - key: all-users
+    name: All Users
+    match_type: ALL_MATCH_TYPE
+```
+
+Each file identified for use by Flipt represents the contents of a single namespace.
+Multiple namespaces can be defined across multiple files.
+You can organize these files how you like in your target directory.
+
+<Tip>
+  By defining different namespaces in different directories, you can leverage
+  features such as GitHub's Codeowners. This gives you authorization mechanisms
+  for managing contributions to Flipt state.
+</Tip>
+
+The file format currently consists of four top-level keys:
+
+```yaml
+version: "1.0" # a version for this file format
+namespace: default # string identifying the resources collective namespace
+flags: [] # [Flag] list of Flag definitions
+segments: [] # [Segment] list of Segment definitions
+```
+
+#### Flags and Variants
+
+<Info>
+  All the resources in Flipt are explained in more depth in the
+  [Concepts](/concepts) documentation page.
+</Info>
+
+The flags section is where your `Flag`, `Variant` and `Rule` definitions live.
+A Flag has many variants and many rules.
+
+```yaml
+flags:
+  - key: string
+    name: string
+    description: string
+    enabled: bool
+    variants: [] # [Variant] a list of variants
+    rules: [] # [Rule] a list of rules
+```
+
+Variants belong to flags and they are nested within each flag definition in a list under the key `variants`.
+A variant consists of a `key`, `name` and optional `description`.
```

Comment:
```suggestion
```

Prob just delete this since they also contain an `attachment` (optional) and may evolve in the future which would mean we'd have to update this again. I think pointing users to our Concepts docs are enough, we don't need to specify each of the attributes.

Reply:
Nice, yeah I wasn't sure how much depth to go into these things here.